src/media-groups.js

```diff
@@ -395,36 +444,19 @@ export const initialize = {
       if (!groups[groupId]) {
         groups[groupId] = [];
       }
-
-      // List of playlists that have an AUDIO attribute value matching the current
-      // group ID
-      const groupPlaylists = playlists.filter(playlist => {
-        return playlist.attributes[type] === groupId;
-      });
-
       for (const variantLabel in mediaGroups[type][groupId]) {
         let properties = mediaGroups[type][groupId][variantLabel];
 
-        // List of playlists for the current group ID that do not have a matching uri
-        // with this alternate audio variant
-        const unmatchingPlaylists = groupPlaylists.filter(playlist => {
-          return playlist.resolvedUri !== properties.resolvedUri;
-        });
-
-        // If there are no playlists using this audio group other than ones
-        // that match it's uri, then the playlist is audio only. We delete the resolvedUri
-        // property here to prevent a playlist loader from being created so that we don't have
-        // both the main and audio segment loaders loading the same audio segments
-        // from the same playlist.
-        if (!unmatchingPlaylists.length && groupPlaylists.length) {
-          delete properties.resolvedUri;
-        }
-
         let playlistLoader;
 
-        // if vhs-json was provided as the source, and the media playlist was resolved,
-        // use the resolved media playlist object
-        if (sourceType === 'vhs-json' && properties.playlists) {
+        if (audioOnlyMaster) {
+          logger_(`AUDIO group '${groupId}' label '${variantLabel}' is a master playlist`);
+          properties.masterPlaylist = true;
```

Comment:
When reading from the top I was a bit confused until I saw that `masterPlaylist` was a boolean. Maybe `isMasterPlaylist`?